contracts/0.8.9/WithdrawalQueue.sol

```diff
@@ -366,4 +406,16 @@ abstract contract WithdrawalQueue is AccessControlEnumerable, PausableUntil, Wit
             revert RequestAmountTooLarge(_amountOfStETH);
         }
     }
+
+    /// @notice returns claimable ether under the request with _requestId.
+    function _getClaimableEther(uint256 _requestId, uint256 _hint) internal view returns (uint256) {
+        if (_requestId == 0 || _requestId > getLastRequestId()) revert InvalidRequestId(_requestId);
+
+        if (_requestId > getLastFinalizedRequestId()) return 0;
+
+        WithdrawalRequest storage request = _getQueue()[_requestId];
```

Comment:
I guess we can load to `memory` the whole structure without significant gas losses

anyway, we read `claimed` right here (slot 1) and `cumulativeStETH` (slot 0) inside `_calculateClimableEther`
so we may pass only `memory` struct to `_calculateClaimableEther` without the separate `_requestId`

Reply:
But we ll need the requestid to check hint validity, I believe, inside `_calculateClaimableEth`

Reply:
Ah, I see, the structure doesn't contain it, of course